src/DynamoCoreWpf/NodeViewCustomization/NodeViewCustomizations/FunctionNodeViewCustomization.cs

```diff
@@ -93,17 +93,31 @@ private void EditCustomNodeProperties()
 
             if (args.Success)
             {
-                CustomNodeWorkspaceModel ws;
-                model.CustomNodeManager.TryGetFunctionWorkspace(
-                    functionNodeModel.Definition.FunctionId,
-                    DynamoModel.IsTestMode,
-                    out ws);
-                ws.SetInfo(args.Name, args.Category, args.Description);
-
-                if (!string.IsNullOrEmpty(ws.FileName))
-                {
-                    ws.Save(ws.FileName, false, dynamoViewModel.EngineController);
-                }
+                SerializeCustomNodeWorkspaceWithNewInfo(args, dynamoViewModel, functionNodeModel);
+            }
+        }
+
+        /// <summary>
+        /// Serialize and update dyf based on FunctionNamePromptEventArgs
+        /// </summary>
+        /// <param name="args">FunctionNamePromptEventArgs which contains updated dyf info</param>
+        /// <param name="dynamoViewModel">Dynamo View Model</param>
+        /// <param name="functionNodeModel">Custom Node</param>
+        internal void SerializeCustomNodeWorkspaceWithNewInfo(FunctionNamePromptEventArgs args, DynamoViewModel dynamoViewModel, Function functionNodeModel)
+        {
+            CustomNodeWorkspaceModel ws;
+            dynamoViewModel.Model.CustomNodeManager.TryGetFunctionWorkspace(
+                functionNodeModel.Definition.FunctionId,
+                DynamoModel.IsTestMode,
+                out ws);
+            ws.SetInfo(args.Name, args.Category, args.Description);
+
+            if (!string.IsNullOrEmpty(ws.FileName))
+            {
+                // Construct a temp WorkspaceViewModel based on for serializing
+                WorkspaceViewModel temp = new WorkspaceViewModel(ws, dynamoViewModel);
```

Comment:
do you think there is any risk to this? side-effects etc?

Reply:
do you have time to profile this with dot memory? As we know from @andydandy74 's use of the migration extension - there is definitely a memory leak somewhere, likely having to do with workspaces. I guess it's not that common that someone would edit their custom node properties repeatedly in one session though.

Reply:
@QilongTang responded to this above, the custom node might not be the current workspace thats open when the property editing occurs, and apparently it might not even be loaded... which seems odd, but we do lazy loading in some cases.

Reply:
I guess the workspaceViewModel might not exist unless the custom node is opened... maybe we could try that first, and then fallback to making a new workspaceViewModel... I guess I'm just worried what consequence there might be for making two workspaceViewModels that point to the same model.

Reply:
I missed that comment then. CurrentSpaceViewModel will refer to HomeWorkpsaceViewModel or something. I am not sure why it will be null or something in that case.

Reply:
@ramramps The currentWorksapceViewModel point to the homeworkspace containing the custom node so calling save there will update the dyn only, user is trying to update the dyf which may or may not be opened at all

Reply:
makes sense. then I think this is the best one for now.